text/metrics/0117-min-max-last-sum-count-aggregator.md

```diff
@@ -0,0 +1,192 @@
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  //
+  // Value of 0 indicates that the timestamp is unspecified. In that case the timestamp
+  // may be decided by the backend.
+  fixed64 start_time_unix_nano = 2;
+
+  // time_unix_nano is the moment when this value was recorded.
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  fixed64 time_unix_nano = 3;
+
+  // last_time_unix_nano is the moment when the latest value was captured during
+  // the interval described by [start_time_unix_nano, time_unix_nano].
+  fixed64 last_time_unix_nano = 4;
+
+  // last_int64 is the last value as an integer.  This field MUST be set when the 
+  // metric descriptor type is MINMAXLASTSUMCOUNT_INT64.
+  int64 last_int64 = 5;
+
+  // min_int64 is the minimum value as an integer, when set.  If this field is
+  // not set and last_int64 is set, the minimum value equals the value of last_int64.
+  int64 min_int64 = 6;
+
+  // max_int64 is the maximum value as an integer, when set.  If this field is
+  // not set and last_int64 is set, the maximum value equals the value of last_int64.
+  int64 max_int64 = 7;
+
+  // last_double is the last value as an integer.  This field MUST be set when the 
+  // metric descriptor type is MINMAXLASTSUMCOUNT_DOUBLE.
+  double last_double = 8;
+
+  // min_double is the minimum value as an integer, when set.  If this field is
+  // not set and last_double is set, the minimum value equals the value of last_double.
+  double min_double = 9;
+
+  // max_double is the maximum value as an integer, when set.  If this field is
+  // not set and last_double is set, the maximum value equals the value of last_double.
+  double max_double = 10;
+
+  // sum is the sum of individual values captured during the interval
+  // described by [start_time_unix_nano, time_unix_nano].  This is a floating
+  // point value even when the metric descriptor type is MINMAXLASTSUMCOUNT_INT64
+  // because it may derived from sample data with a non-integer multiplier.
+  //
+  // When this value is not set, it may be presumed to equal the value of
+  // last_int64 or last_double, whichever field is set, times the value of count.
+  double sum = 11;
+
+  // count is the number of individual values captured during the interval
+  // described by [start_time_unix_nano, time_unix_nano].  this is a floating
+  // point value because it may derived from sample data with a non-integer
+  // multiplier.
+  //
+  // This MUST be omitted when the value is exactly 1.0.  When this value
+  // is not set, it MUST be assumed to equal 1.0.
+  double count = 12;
+}
+```
+
+This message type is specified so that:
+
+- the count can be omitted when it equals 1.0
+- the minimum and maximum can be omitted when they are equal to the last value
+- the sum value can omitted when it equals the last value times the count.
```

Comment:
It seems like there could be implementations of this proto that are not able to distinguish between unset and a zero value right? If that is the case, it might not be a good idea to allow omission as it could introduce ambiguity.

Reply:
I simplified this proposal and clarified that we do not need to distinguish between omitted and zero values.